xchange-coingi/src/main/java/org/knowm/xchange/coingi/service/CoingiTradeService.java

```diff
@@ -0,0 +1,177 @@
+package org.knowm.xchange.coingi.service;
+
+import static org.knowm.xchange.dto.Order.OrderType.BID;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Date;
+import org.knowm.xchange.Exchange;
+import org.knowm.xchange.coingi.CoingiAdapters;
+import org.knowm.xchange.coingi.CoingiErrorAdapter;
+import org.knowm.xchange.coingi.dto.CoingiException;
+import org.knowm.xchange.coingi.dto.trade.*;
+import org.knowm.xchange.currency.CurrencyPair;
+import org.knowm.xchange.dto.Order;
+import org.knowm.xchange.dto.trade.LimitOrder;
+import org.knowm.xchange.dto.trade.MarketOrder;
+import org.knowm.xchange.dto.trade.OpenOrders;
+import org.knowm.xchange.dto.trade.UserTrades;
+import org.knowm.xchange.exceptions.NotYetImplementedForExchangeException;
+import org.knowm.xchange.service.trade.TradeService;
+import org.knowm.xchange.service.trade.params.CancelOrderByIdParams;
+import org.knowm.xchange.service.trade.params.CancelOrderParams;
+import org.knowm.xchange.service.trade.params.TradeHistoryParamPaging;
+import org.knowm.xchange.service.trade.params.TradeHistoryParams;
+import org.knowm.xchange.service.trade.params.orders.DefaultOpenOrdersParamCurrencyPair;
+import org.knowm.xchange.service.trade.params.orders.OpenOrdersParams;
+
+public class CoingiTradeService extends CoingiTradeServiceRaw implements TradeService {
+  public CoingiTradeService(Exchange exchange) {
+    super(exchange);
+  }
+
+  @Override
+  public OpenOrders getOpenOrders() throws IOException {
+    return getOpenOrders(createOpenOrdersParams());
+  }
+
+  @Override
+  public OpenOrders getOpenOrders(OpenOrdersParams params) throws IOException {
+    CoingiGetOrderHistoryRequest orderHistoryRequest = new CoingiGetOrderHistoryRequest();
+    orderHistoryRequest.setStatus(0);
+    orderHistoryRequest.setPageNumber(1);
+    orderHistoryRequest.setPageSize(50);
+    try {
+      CoingiOrdersList list = getCoingiOrderHistory(orderHistoryRequest);
+      return CoingiAdapters.adaptOpenOrders(list);
+    } catch (CoingiException e) {
+      throw CoingiErrorAdapter.adapt(e);
+    }
+  }
+
+  @Override
+  public String placeMarketOrder(MarketOrder order) throws IOException {
+    throw new NotYetImplementedForExchangeException();
+  }
+
+  @Override
+  public String placeStopOrder(org.knowm.xchange.dto.trade.StopOrder stopOrder)
+      throws java.io.IOException {
+    throw new NotYetImplementedForExchangeException();
+  }
+
+  @Override
+  public String placeLimitOrder(LimitOrder order) throws IOException {
+    CoingiPlaceLimitOrderRequest request =
+        new CoingiPlaceLimitOrderRequest()
+            .setCurrencyPair(CoingiAdapters.adaptCurrency(order.getCurrencyPair()))
+            .setOrderType(order.getType().equals(BID) ? 0 : 1)
+            .setPrice(order.getLimitPrice())
+            .setVolume(order.getRemainingAmount());
+
+    String response;
+    try {
```

Comment:
the try / catch CoingiException  hit not applied

Reply:
done, thanks for spotting this